### PR TITLE
Read CLI options from the input file

### LINKED ISSUE #5: refactor get_metadata function
The current implementation is not the cleanest one. Think of a cleaner implementation.

### LINKED ISSUE #8: read cli options from input file
Read all the CLI options from the config file and use them as default values.

### PR DESCRIPTION
**Pull request by mustafasoylu**
- Read CLI options from the input file
Options that are directly given to CLI have higher priority than the options in the input file
- Add a command that writes CLI options to the input files with prompts
- refactor load functions in core
- add badges to the readme
- update readme for the new command
resolves #5 
resolves #8

### CODE REVIEW

File: src/somesy/cli/validators.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    if cli_options.pyproject_file is not None:
+        validated_inputs["no_sync_pyproject"] = False
+
+    # at least one of the sync options must be enabled
+    if validated_inputs["no_sync_cff"] and validated_inputs["no_sync_pyproject"]:
+        logger.warning("There should be at least one file to sync.")
+        typer.Exit(code=0)
+
+    logger.debug(f"validated inputs: {validated_inputs}")
+    return validated_inputs
```
Comment (apirogov) [inline src/somesy/cli/validators.py:59]: Is this method really needed?

You probably should just update your empty/loaded config with all non-default values passed in the CLI, similarly as described here https://github.com/pydantic/pydantic/discussions/3139#discussioncomment-4797649

i.e. turn the cli options model into `dict` with `exclude_defaults=True`, and use `update` on your file options model.

Comment (apirogov) [reply src/somesy/cli/validators.py:59]: So I'd streamline that to:

* having your Options class with the defaults
* updating it as described above from a detected somesy config file
* updating it the same way from cli arguments, if given
* running the validation on the result

Or something like that.

Comment (mustafasoylu) [reply src/somesy/cli/validators.py:59]: I realize it already has a pydantic model so, this dataclass and validation function is completely redundant. I will update it accordingly.

Comment (mustafasoylu) [reply src/somesy/cli/validators.py]: @apirogov updated accordingly with [this commit](https://github.com/Materials-Data-Science-and-Informatics/somesy/pull/11/commits/422938d9abe22cc93b94e31e6419a0a87ef3c090)

**Review by apirogov — CHANGES_REQUESTED** [review]
As discussed, looks fine, except for me wondering why the merged config is a plain dict after merging ;)

**Comment by mustafasoylu** [comment]
@apirogov implemented as pydantic class object

**Pull request merged**